### PR TITLE
Generate VSIX artifact with GitHub Actions

### PR DESCRIPTION
**Pull request by acidicMercury8**
Changes proposed in this pull request:

- Enabled GitHub Actions for building Hexa VSCode bundle

Related issue:

- #1 

Agreements

- [x] I agree to follow this project's [__Contributing Guidelines__](https://github.com/hexalang/.github/blob/kawaii/CONTRIBUTING.md)
- [x] I agree to follow this project's [__Code of Conduct__](https://github.com/hexalang/.github/blob/kawaii/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "bugs": {
         "url": "https://github.com/hexalang/hexa-vscode-bundle/issues"
     },
-    "dependencies": {
+    "devDependencies": {
+        "vsce": "^2.9.1"
```
Comment (PeyTy) [inline package.json]: Can you create a completely separate `package.json` for `vsce`? Cause this one is used by the extension itself @acidicMercury8

File: .github/workflows/building-bundle.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  build-all:
+    name: 'Building'
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ windows-latest, ubuntu-latest, macos-latest ]
```
Comment (PeyTy) [inline .github/workflows/building-bundle.yml]: No need for matrix builds, no native code is used in extension. Just keep `windows-latest` @acidicMercury8

File: .github/workflows/building-bundle.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+
+    - name: Install dependencies
+      run: |
+        npm clean-install
```
Comment (PeyTy) [inline .github/workflows/building-bundle.yml]: This is not required, `npX` command simply uses global node cache. Extension also works without any node_modules dependencies.

Comment (PeyTy) [reply .github/workflows/building-bundle.yml]: I.e. `npx vsce` will run `vsce` ***without*** installation, that's the `npx` command feature

File: package.json (hunk 1 above)
Comment (PeyTy) [reply package.json]: Update: check my comment about npx

File: .github/workflows/building-bundle.yml (hunk 2 above)
Comment (acidicMercury8) [reply .github/workflows/building-bundle.yml]: @PeyTy okay, fixed

**Review by PeyTy — APPROVED** [review]

**Pull request merged**